transcom/mymove
# B 19830 int and B 19843 INT

## [B-19830](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19830) and [B-19843](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19843)

## Summary
Since Landan is going to be away on military orders, I've gone ahead and taken B-19843 and B-19830 and compressed them into a single PR to allow for an easier time managing and testing the backlogs together. As a result, this both replaces "Claimable Expenses" with "Claimed Expenses", and introduces a new subheading next to Trusted agents, reading "Trusted Agent Requires POA or Letter of Authorization".

Additionally, I encountered a bug where the SSW/AOA cannot be generated or downloaded when there are expense items without a type or cost associated with them (They can be saved like this, if a user starts filling one out and then closes the application). This includes a nil check at the start of the expense tallying loop that will skip over any items with a nil type or cost.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

1. Access the application as a customer, create and submit a PPM with an advance.
2. Log in as a service counselor, approve the PPM and the Advance.
3. Back as the original customer, click "Add PPM Documents", fill out the weight ticket information, proceed.
4. On the next page, add an expense.
5. Before filling out the information, log out or leave back to the application home page.
6. Click "Add PPM Documents" again. The empty expense should be present.
7. Return to the move page. Download your AOA Packet, ensure that the file downloads correctly and the copy changes match those in the screenshot below.

### Frontend


### Backend


## Screenshots
<img width="564" alt="image" src="https://github.com/transcom/mymove/assets/148906585/104c47fa-dbac-409d-b7d5-2477e292c32e">
